lib/rucio/common/stomp_utils.py

```diff
@@ -15,145 +15,415 @@
+        nonssl_port = config_get_int(config_section, 'nonssl_port', default=0, raise_exception=False)
+        timeout = config_get_float(config_section, 'timeout', default=None, raise_exception=False)
+        heartbeats = tuple(config_get_list(config_section, 'heartbeats', default=[0., 0.], raise_exception=False))
+        reconnect_attempts = config_get_int(config_section, 'reconnect_attempts', default=100)
+        if use_ssl and (key_file is None or cert_file is None):
+            self._logger(logging.ERROR, "If use_ssl is True in config you must provide both 'ssl_cert_file' "
+                                        "and 'ssl_key_file'")
+            raise RuntimeError("If use_ssl is True in config you must provide both 'ssl_cert_file' and 'ssl_key_file'")
+        if not use_ssl and (username is None or password is None or nonssl_port == 0):
+            self._logger(logging.ERROR, "If use_ssl is False in config you must provide "
+                                        "'username', 'password' and 'nonssl_port'")
+            raise RuntimeError("If use_ssl is False in config you must provide "
+                               "'username', 'password' and 'nonssl_port'")
+        return StompConfig(brokers=self._resolve_host_and_port(brokers, port if use_ssl else nonssl_port),
+                           use_ssl=use_ssl,
+                           port=port, vhost=vhost,
+                           destination=destination, key_file=key_file, cert_file=cert_file,
+                           username=username, password=password, nonssl_port=nonssl_port,
+                           reconnect_attempts_max=reconnect_attempts, timeout=timeout, heartbeats=heartbeats)
+
+    def _resolve_host_and_port(self, fqdns: str | list[str], port: int) -> list[tuple[str, int]]:
+        """
+        Resolve host and port.
+
+        Args:
+            fqdns (str | list[str]): fully qualified domain name(s)
+            port (int): port
+
+        Returns:
+            list[tuple[str, int]]: list of (host, port) tuples.
+        """
+        if isinstance(fqdns, str):
+            fqdns = fqdns.split(',')
+
+        hosts_and_ports = []
+        for fqdn in fqdns:
+            try:
+                addrinfos = socket.getaddrinfo(fqdn.strip(), port, socket.AF_INET, 0, socket.IPPROTO_TCP)
+            except socket.gaierror as exc:
+                self._logger(logging.ERROR, "[broker] Cannot resolve domain name %s (%s)", fqdn.strip(), str(exc))
+                continue
+
+            hosts_and_ports.extend(addrinfo[4] for addrinfo in addrinfos)
+        if not hosts_and_ports:
+            self._logger(logging.WARNING, "[broker] No resolved brokers")
+        return hosts_and_ports
+
+    def _is_stalled(self, conn: Connection) -> bool:
+        """
+        Determine if a connection is stalled.
+
+        Args:
+            conn (Connection): The Connection object
```

Comment:
Please avoid specifying the argument types in the docstring if they are already present as type annotations; there's no need to duplicate this information and it can cause mismatches between the type annotation and the type specified in the docstring if changes are made down the line